mono/utils/mono-error-internals.h

```diff
@@ -15,7 +15,10 @@ typedef struct {
 	const char *member_name;
 	const char *exception_name_space;
 	const char *exception_name;
-	MonoClass *klass;
+	union {
+		MonoClass *klass; /* used by type or field load errors and generic error specified by class */
```

Comment:
More than documenting what they mean, for union what we want is which condition trigger which case. Like, for example:
```
uint32_t instance_handle; /* only if error_code == MONO_ERROR_EXCEPTION_INSTANCE */
```